src/rendering/renderers/pool/PoolRenderer.ts

```diff
@@ -415,7 +424,7 @@ export class PoolRenderer extends BaseRenderer {
         const { torusstyle } = style
```

Comment:
I think there should be couple simple changes in this file
1. You should read texture from `context.textures.get(...)` and add it into `coveredNameStyle.texture`
```
const { coveredNameStyle } = style
coveredNameStyle.texture = context.textures.get(MAP_TO_TEXTURES[pool.level])

const [coveredName, coveredNameState] = this.get('cover'+title, () => GraphicUtils.createCoveredText(
            title,
            [x + coverpadding, paddingTop],
            coveredNameStyle,
        ), [pool.level])
```
To implement that right you should add dependency to `this.get(...)` function in order to re-init the graphic

2. You should add logic to `createCoveredText` function to apply texture it was passed to the function via `styles`

3. All textures should be to TextureStorage

4. Gradient angle is setted by start and end line coordinates